Modules/Order/Http/Controllers/OrderController.php

```diff
@@ -23,159 +21,318 @@ public function __construct(OrderService $orderService)
         $this->orderService = $orderService;
     }
 
+    /**
+     * Show all order for logged user
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
     public function index(Request $request): JsonResponse
     {
-        $user = auth('sanctum')->user();
+        $user = auth()->user();
+        $role = $user->role;
+        if($role==='brand')
+        {
+            if ($user->cannot('viewAnyBrand', Order::class)) {
+                return response()->json([
+                    'res' => false,
+                    'msg' => 'User is not authorized !',
+                    'data' => ""
+                ]);
+            }
+        }
+        if($role==='retailer')
+        {
+            if ($user->cannot('viewAnyRetailer', Order::class)) {
+                return response()->json([
+                    'res' => false,
+                    'msg' => 'User is not authorized !',
+                    'data' => ""
+                ]);
+            }
+        }
+
 
         $response = $this->orderService->index($request);
 
         return response()->json($response);
     }
+
     /**
-     * Store a newly created order in storage
+     * Add shipping and billing address for order
      *
-     * @param OrderRequest $request
+     * @param Request $request
      * @return JsonResponse
      */
     public function checkout(Request $request): JsonResponse
     {
-        $user = auth('sanctum')->user();
+        $user = auth()->user();
+        if ($user->cannot('viewAnyRetailer', Order::class)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
 
         $response = $this->orderService->checkout($request);
 
         return response()->json($response);
     }
 
     /**
+     * Update billing address by retailer
+     *
      * @param OrderRequest $request
      * @return JsonResponse
      */
-    public function UpdateBilling(OrderRequest $request): JsonResponse
+    public function updateBilling(OrderRequest $request): JsonResponse
     {
-        $user = auth('sanctum')->user();
-        $request->request->add(['user_id' => $request->user_id]);
-        $response = $this->orderService->updatebilling($request->validated());
+        $user = auth()->user();
+        if ($user->cannot('viewAnyRetailer', Order::class)) {
```

Comment:
For checkout better user another policy rule called "update", viewAny means view and nothing more

Reply:
Done